wordpress-mobile/WordPress-iOS
# Domains Dashboard

Fixes #17184
Fixes #17185
Fixes #17186
Fixes #17187
Fixes #17188

This PR adds the complete version of the Domains Dashboard, aside from a few missing tweaks (see known issues). It implements the following designs:

<img width="754" src="https://user-images.githubusercontent.com/34376330/134760649-a7446b70-18f5-46bb-8f17-de88a3a89510.png">

To test:

Pre-requirements:
- make sure you're using the store sandbox (recommended)
- make sure you have the following:
    - A site with a free plan and no registered domains
    - A site with a free plan and one (or more) registered domains
    - A site with a paid plan that has not yet claimed the free domain
    - A site with a paid plan that has already claimed the free domain

Test:
- Build Run
- Select one of the sites described in the pre-requirements
- Tap on Domains, to access the Domains Dashboard
- Make sure that it looks consistent with the corresponding design (see above)
- Make sure that the "Primary site address" label shows in the correct place
- Make sure the buttons are actionable
- Make sure that the domain purchase flow starts in a modal
- Optionally, make sure that you can actually complete a domain registration successfully
- Repeat all the above steps for every domain listed in the pre-requirements

**KNOWN ISSUES**
- Domains in the domain list do not contain the status (e.g. expires on xxx)
- Modals do not automatically dismiss
- Any precise spacing and sizing will be fixed in a future PR
- This PR does not concern domains search or purchases: any issues in other parts of the flow will be addressed in future PRs


## Regression Notes
1. Potential unintended areas of impact
This PR impacts the new Domains Dashboard and the new domain registration flow

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Test the new flow

3. What automated tests I added (or what prevented me from doing so)
None. Automated tests might be added in a future PR.

PR submission checklist:
